rules/application/kubernetes/audit/kubernetes_audit_hostpath_mount.yml

```diff
@@ -0,0 +1,26 @@
+title: Container With A hostPath Mount Created
+id: 402b955c-8fe0-4a8c-b635-622b4ac5f902
+status: experimental
+description: |
+    Detects creation of a container with a hostPath mount.
+    A hostPath volume mounts a directory or a file from the node to the container. Attackers who have permissions to create a new pod in the cluster may create one with a writable hostPath volume and chroot to escape to the underlying node.
+references:
+    - https://microsoft.github.io/Threat-Matrix-for-Kubernetes/techniques/Writable%20hostPath%20mount/
+    - https://blog.appsecco.com/kubernetes-namespace-breakout-using-insecure-host-path-volume-part-1-b382f2a6e216
+author: Leo Tsaousis (@laripping)
+date: 2024/03/08
+tags:
+    - attack.t1611
+logsource:
+    category: application
+    product: kubernetes
+    service: audit
+detection:
+    selection:
+        verb: create
+        resource: pods
+        hostPath: "*"
```

Comment:
Are you trying to specify the existence of the field or just saying **any** hostPath ?

Reply:
Both. IIRC the way the Kubernetes Elastic Integration works, if these fields are not present in the audit logs ingested, they will not appear as fields in the JSON events. Therefore by specifying this Sigma filtering statement explicitly, the backend will translate it into an additional ELK query filter (with a wildcard value) so that the complete resulting query is refined further, filtering out events  without this field ("exists" behaviour)

In reverse, omitting this Sigma filter, will also omit it from the resulting ELK query thus matching events  with or without this field specified. 

See the unit test `test_ecs_kubernetes_capabilities` which intends to demonstrate this behaviour: 
https://github.com/SigmaHQ/pySigma-backend-elasticsearch/blob/8bc0e35d6476bc36d7ebac622807deeb8167a5de/tests/test_pipelines_kubernetes.py#L41